src/compile/scale/range.ts

```diff
@@ -329,17 +334,18 @@ export function defaultContinuousToDiscreteCount(
  * @param rangeMax end of the range
  * @param cardinality number of values in the output range
  */
-export function interpolateRange(
-  rangeMin: number,
-  rangeMax: number | SignalRefComponent,
-  cardinality: number
-): SignalRefComponent {
-  const step = '(rangeMax - rangeMin) / (cardinality - 1)';
-  return evalOrMakeSignalRefComponent(`sequence(rangeMin, rangeMax + ${step}, ${step})`, {
-    rangeMin,
-    rangeMax,
-    cardinality
-  });
+export function interpolateRange(rangeMin: number, rangeMax: number | SignalRef, cardinality: number): SignalRef {
+  // always return a signal since it's better to compute the sequence in Vega later
+  const f = () => {
+    const rMax = isSignalRef(rangeMax) ? rangeMax.signal : rangeMax;
+    const step = `(${rMax} - ${rangeMin}) / (${cardinality} - 1)`;
+    return `sequence(${rangeMin}, ${rangeMax} + ${step}, ${step})`;
+  };
+  if (isSignalRef(rangeMax)) {
```

Comment:
You should keep `evalOrMakeSignalRefWrapper` (renamed) so you don't need to do this check manually. 

The reason I structure it this way (in the old code) is so that when we support signal for more properties, you don't have to change too much code (as the check for rangeMin / cardinality should come for free too).

Reply:
How would this work now that we don't use a regex to replace things in expressions anymore? Can I do this as a follow up? I'll create an issue.